Android build

ili opened the pull request:
## What does the pull request do?

- Build & run ControlCatalog.Android
- Build & run Avalonia.AndroidTestApplication
- Running ControlCatalog.Desktop on x64

## What is the current behavior?
Build or run fails

## What is the updated/expected behavior with this PR?
Build succeeed, application runs

## How was the solution implemented (if it's not obvious)?
<!--- Include any information that might be of use to a reviewer here. -->
Mostly correct usings & package referencies.
The only point i had to extend `ITopLevelImpl` with `IInputRoot InputRoot { get; }` (previously it used to be set only property)


## Checklist
- [x] Added XML documentation to any related classes?

## Breaking changes
None
<!--- List any breaking changes here. When the PR is merged please add an entry to https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes -->


## Fixed issues
<!--- If the pull request fixes issue(s) list them like this: 
Fixes #123
Fixes #456
-->
Partially fixies #4317

ili committed:
Build & Run ControlCatalog.Android

ili committed:
Merge remote-tracking branch 'origin/master' into android_build

dnfadmin commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=4378) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=4378) before we can accept your contribution.<br/><br/>:x: ili [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=4378)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=4378) it.</sub>

kekekeks commented on src/Avalonia.Controls/Platform/ITopLevelImpl.cs:
```diff
@@ -80,6 +80,11 @@ public interface ITopLevelImpl : IDisposable
+        /// <summary>
+        /// Gets the <see cref="IInputRoot"/> for the toplevel.
+        /// </summary>
+        IInputRoot InputRoot { get; }
```
This changes the platform backend interface, so it involves every backend.

If we are doing that, might as well make it an auto property instead of SetInputRoot + property.

@grokys

ili replied:
Done. As for me it is good solution, even it was already done so for `WindowBaseImpl` & used in `WindowBaseEvents` for ex.

ili committed:
Build & Run Avalonia.AndroidTestApplication

ili committed:
Running ControlCatalog.Desktop on x64

Using x86 target for ControlCatalog.Desktop (skia x86 natives distributed during build)

Commit:
Merge branch 'master' into android_build

wieslawsoltes commented on samples/ControlCatalog.Android/ControlCatalog.Android.csproj:
```diff
@@ -152,8 +156,16 @@
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="HarfBuzzSharp">
```
This should be added as .props file https://github.com/AvaloniaUI/Avalonia/blob/master/build/HarfBuzzSharp.props

wieslawsoltes commented on samples/ControlCatalog.Android/ControlCatalog.Android.csproj:
```diff
@@ -152,8 +156,16 @@
+    <PackageReference Include="HarfBuzzSharp">
+      <Version>2.6.1.5</Version>
+    </PackageReference>
+    <PackageReference Include="SkiaSharp">
```
This should be added as .props file https://github.com/AvaloniaUI/Avalonia/blob/master/build/SkiaSharp.props

wieslawsoltes commented on src/Android/Avalonia.AndroidTestApplication/Avalonia.AndroidTestApplication.csproj:
```diff
@@ -148,10 +152,18 @@
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="HarfBuzzSharp">
```
https://github.com/AvaloniaUI/Avalonia/blob/master/build/SkiaSharp.props
https://github.com/AvaloniaUI/Avalonia/blob/master/build/HarfBuzzSharp.props

ili committed:
Merge remote-tracking branch 'origin/master' into android_build

ili committed:
Using .props instead of direct package reference

ili committed:
Merge branch 'master' into android_build

ili committed:
Merge branch 'master' into android_build

ili committed:
Convert ITopLevelImpl.InputRoot to r/w property

ITopLevelImpl.SetInputRoot - removed

ili committed:
Merge branch 'master' into android_build

ili committed:
Merge branch 'master' into android_build

ili committed:
Merge branch 'master' into android_build

ili committed:
Merge branch 'master' into android_build

ili committed:
Merge branch 'master' into android_build

ili committed:
KitKat for Android demo apps as minSdk

ili committed:
Merge branch 'master' into android_build

ili committed:
Merge branch 'master' into android_build

danwalmsley commented:
@workgroupengineering iOS is actively being worked on and Android will follow soon.. so this is coming... but if anyone wants to help whip android backend please get in touch directly via team@avaloniaui.net so we can discuss on a call how to proceed.

Need to get the api changes here approved before merging this.

ili commented:
@danwalmsley this is really awesome news! We are all expecting Avalonia to be Flutter-killer :)))

maxkatz6 commented:
What is the state of this PR?
I just wonder if it can be used for some simplest applications even w/ bugs and w/o stability.
Because current android backend is completely broken.

jp2masa commented:
I don't know what's planned for the Android backend or if there's any work in progress, but this PR hasn't been updated for some time and I was making some experiments with the backend today, would you be interested in a PR with the minimal changes needed to at least be able to run? (Control Catalog is really slow)
